src/graaflib/directed_graph.h

```diff
@@ -8,6 +8,13 @@ namespace graaf {
 template <typename VERTEX_T, typename EDGE_T>
 class directed_graph final
     : public graph<VERTEX_T, EDGE_T, graph_spec::DIRECTED> {
+ public:
+  [[nodiscard]] std::size_t get_vertex_outdegree(
+      vertex_id_t vertex_id) const noexcept;
+
+  [[nodiscard]] std::size_t get_vertex_indegree(
+      vertex_id_t vertex_id) const noexcept;
+
```

Comment:
Should it also be possible to compute `get_vertex_degree` for a vertex in a directed graph and get the sum of the in and out vertices?

If we are splitting these methods into free functions we could have the following three functions:

```c++
[[nodiscard]] std::size_t get_vertex_degree(const graaf::graph<V, E, S>&, vertex_id_t);
[[nodiscard]] std::size_t get_vertex_outdegree(const graaf::graph<V, E, S>&, vertex_id_t);
[[nodiscard]] std::size_t get_vertex_indegree(const graaf::graph<V, E, S>&, vertex_id_t);
```

Here the graph parameter has the type of the derived-from class, such that we can accept directed as well as undirected graphs. You could then dispatch to different logic inside each function, i.e.

```c++
template <typename V, typename E, graph_spec SPEC>
std::size_t get_vertex_degree(const graaf::graph<V, E, SPEC> graph, vertex_id_t vertex_id) {
 if constexpr (SPEC == graph_spec::DIRECTED) {
    return get_vertex_outdegree(graph, vertex_id) + get_vertex_indegree(graph, vertex_id);
  } 

  if constexpr (SPEC == graph_spec::UNDIRECTED) {
    // do undirected logic
  }

  // Should never reach this
  std::abort();
}
```

Reply:
> Should it also be possible to compute get_vertex_degree for a vertex in a directed graph and get the sum of the in and out vertices?

I don't think there is a mathematical definition for degree of directed graphs, but if there was one, this would probably be it. So we can definitely go with it.

> If we are splitting these methods into free functions

I don't have a strong opinion here. My rationale was to align them to `get_neighbors`, which is defined as class method and eats a `vertex_id` as well. But I see that they are dealing with a different level of abstraction. Happy to outsource them into free functions. Will do this in a separate PR to enable comparing them side-by-side.

> such that we can accept directed as well as undirected graphs

What would a call of `get_vertex_in/outdegree` on an undirected graph do?

1. Throw exception
2. Return 0 
3. Return the same value as `get_vertex_degree`

I would vote for the third option, since an undirected edge can be seen as both an outgoing and incoming edge.